front/src/common/InfraSelector/InfraSelectorEditionActionsBarDelete.tsx

```diff
@@ -17,12 +18,15 @@ export default function InfraSelectorEditionActionsBarDelete({
   const { t } = useTranslation('infraManagement');
   const dispatch = useDispatch();
 
-  const [deleteInfra] = osrdEditoastApi.useDeleteInfraByIdMutation();
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const [deleteInfra] = osrdEditoastApi.endpoints.deleteInfraById.useMutation();
```

Comment:
I think you could directly use the isLoading boolean given by rtk query instead of creating a new state.
I tested with this solution and it seems to work the same
```suggestion
  const [deleteInfra, { isLoading: isDeleting }] = osrdEditoastApi.endpoints.deleteInfraById.useMutation();
```

Reply:
Wow great idea ! :D Thanks